driver/others/dynamic_power.c

```diff
@@ -52,6 +52,9 @@ static gotoblas_t *get_coretype(void) {
 	if (__builtin_cpu_supports ("arch_3_1") && __builtin_cpu_supports ("mma"))
 		return &gotoblas_POWER10;
 #endif
+	/* Fall back to the POWER9 implementation if the toolchain is too old or the MMA feature is not set */
```

Comment:
Thanks  @antonblanchard for correcting the check for ld version.

If the toolchain is old, I think we might get error like this. 
       error: cpu ‘power10’ is an invalid argument to builtin ‘__builtin_cpu_is’
In which distro are you seeing this?  I remember we added this P10 check specifically to work for fedora 32, which has gcc10.2 and ld 2.34.